dlc-manager/src/sub_channel_manager.rs

```diff
@@ -2741,8 +2824,64 @@ where
                         .map_err(|e| Error::InvalidState(format!("{:?}", e)))?;
                 }
                 SubChannelState::Confirmed(a) => {
+                    self.ln_channel_manager
+                        .with_useable_channel_lock(&channel.channel_id, peer_id, |channel_lock| {
+                            let dlc_channel_id =
+                                channel.get_dlc_channel_id(0).ok_or_else(|| {
+                                    Error::InvalidState("Could not get dlc channel id".to_string())
+                                })?;
+                            let dlc_channel = get_channel_in_state!(
+                                self.dlc_channel_manager,
+                                &dlc_channel_id,
+                                Signed,
+                                None::<PublicKey>
+                            )?;
+                            let contract = get_contract_in_state!(
+                                self.dlc_channel_manager,
+                                &dlc_channel
+                                    .get_contract_id()
+                                    .expect("Signed contract should have a contract id"),
+                                Signed,
+                                None::<PublicKey>
+                            )?;
+                            let offered_channel = OfferedChannel {
+                                offered_contract_id: contract.accepted_contract.offered_contract.id,
+                                temporary_channel_id: dlc_channel.temporary_channel_id,
+                                party_points: dlc_channel.own_points,
+                                per_update_point: dlc_channel.own_per_update_point,
+                                offer_per_update_seed: channel.per_split_seed,
+                                is_offer_party: true,
+                                counter_party: dlc_channel.counter_party,
+                                // TODO(tibo): use value from original offer
+                                cet_nsequence: CET_NSEQUENCE,
+                            };
+                            self.ln_channel_manager.set_funding_outpoint(
+                                channel_lock,
+                                &a.ln_rollback.funding_outpoint,
+                                a.ln_rollback.channel_value_satoshis,
+                                a.ln_rollback.value_to_self_msat,
+                            );
+                            self.dlc_channel_manager.get_store().upsert_channel(
+                                Channel::Offered(offered_channel),
+                                Some(Contract::Offered(
+                                    contract.accepted_contract.offered_contract,
+                                )),
+                            )?;
```

Comment:
Should we resend the offer too? In theory we don't know if the counterparty ever got it, right?

Reply:
If you're in `Confirm` state then you must have received an `Accept` message which means your counter party did receive the offer.